include/mapped_soa.h

```diff
@@ -85,6 +85,27 @@ class MappedSoA {
     key_to_index_.erase(key);
   }
 
+  void copy(size_t index0, size_t index1) {
+    const size_t size_val = size();
+    if (size_val <= index0 || size_val <= index1) {
```

Comment:
I like putting the check inside the assert. At runtime people should make sure they know the ranges of their arrays and we can skip the branch here.

Reply:
Actually, ignore this comment. I forgot I used to do it but changed this. Sorry!

Reply:
:+1: